groupease-server/src/main/java/io/github/groupease/group/GroupDao.java

```diff
@@ -0,0 +1,66 @@
+package io.github.groupease.group;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+
+import static java.util.Objects.requireNonNull;
+
+import io.github.groupease.model.Group;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+
+public class GroupDao {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private final EntityManager entityManager;
+
+    /**
+     * Injectable constructor.
+     *
+     * @param entityManager
+     */
+    @Inject
+    public GroupDao(@Nonnull EntityManager entityManager)
+    {
+        this.entityManager = requireNonNull(entityManager);
+    }
+
+    public List<Group> listInChannel(long channelId) {
```

Comment:
If we put `@Timed` annotations on all non-private methods in Guice-created classes, then we can gather runtime statistics and metrics on them.  This could be good data to present for our project.